pkg/networkmanager/wireguard.jsx

```diff
@@ -0,0 +1,192 @@
+    }, [privateKey]);
+
+    function onSubmit() {
+        function createSettingsObj() {
+            return {
+                connection: {
+                    id: `con-${iface}`,
+                    interface_name: iface,
+                    type: 'wireguard'
+                },
+                wireguard: {
+                    private_key: privateKey.trim(),
+                    listen_port: +listenPort,
+                    peers,
+                },
+                ipv4: {
+                    addresses: [[addresses.split('/')[0], addresses.split('/')[1], '0.0.0.0']], // TODO: fix gateway
+                    method: 'manual'
+                }
+            };
+        }
+
+        dialogSave({
+            model,
+            settings: createSettingsObj(),
+            onClose: Dialogs.close,
+            setDialogError
+        });
+    }
+
+    return (
+        <NetworkModal title={_("WireGuard settings")} onSubmit={onSubmit} dialogError={dialogError} idPrefix={idPrefix}>
+            <Name idPrefix={idPrefix} iface={iface} setIface={setIface} />
+            <FormGroup label={_('Private key')} fieldId={idPrefix + '-private-key-input'}>
+                <InputGroup>
+                    <TextInput id={idPrefix + '-private-key-input'} value={privateKey} onChange={(_, val) => setPrivateKey(val)} />
+                    <Button variant='secondary'
+                        onClick={async () => {
+                            const key = await cockpit.script("wg genkey");
+                            setPrivateKey(key.trim());
+                        }}
+                    >
+                        Generate private key
+                    </Button>
+                </InputGroup>
+            </FormGroup>
+            <FormGroup label={_('Public key')}>
+                <ClipboardCopy isReadOnly>{publicKey}</ClipboardCopy>
+            </FormGroup>
+            <FormGroup label={_('Listen port')} fieldId={idPrefix + '-listen-port-input'}>
+                <TextInput id={idPrefix + '-listen-port-input'} placeholder='Random port selected when unspecified' type='number' onChange={(_, val) => { setListenPort(val) }} />
+            </FormGroup>
+            <FormGroup label={_('Addresses')} fieldId={idPrefix + '-addresses-input'}>
+                <TextInput id={idPrefix + '-addresses-input'} onChange={(_, val) => { setAddresses(val) }} placeholder='E.g. 10.0.0.1/24' />
+            </FormGroup>
+            <FormFieldGroup
+                header={
+                    <FormFieldGroupHeader
+                        titleText={{ text: _("Peers") }}
+                        actions={
+                            <Button
+                                variant='secondary'
+                                onClick={() => setPeers(peers => [...peers, { publicKey: '', endpoint: '', allowedIps: '' }])}
+                            >
+                                {_("Add peer")}
+                            </Button>
```

Comment:
These 2 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19024-20230627-160728-be631904-fedora-38-devel-expensive//Coverage/lcov/github-pr.diff.gcov.html)